src/prefect/tasks/dbt/dbt_cloud_utils.py

```diff
@@ -0,0 +1,95 @@
+from prefect.engine.signals import FAIL
+import requests
+from time import sleep
+
+# dbt Cloud Trigger Job API -> https://docs.getdbt.com/dbt-cloud/api-v2#operation/triggerRun
+__DBT_CLOUD_TRIGGER_JOB_API_ENDPOINT_V2 = (
+    "https://cloud.getdbt.com/api/v2/accounts/{accountId}/jobs/{jobId}/run/"
+)
+
+# dbt Cloud Get Run API -> https://docs.getdbt.com/dbt-cloud/api-v2#operation/getRunById
+__DBT_CLOUD_GET_RUN_API_ENDPOINT_V2 = (
+    "https://cloud.getdbt.com/api/v2/accounts/{accountId}/runs/{runId}/"
+)
+
+
+def trigger_job_run(
+    account_id: int, job_id: int, token: str, cause: str, additional_args: dict
+) -> dict:
+    """
+    Trigger a dbt Cloud job run
+
+    Args:
+        - account_id (int): dbt Cloud account ID
+        - job_id (int): dbt Cloud job ID
+        - token (string): dbt Cloud token
+        - cause (string): the reason describing why the job run is being triggered
+        - additional_args (dict): additional information to pass to the Trigger Job Run API
+
+    Returns:
+        - The trigger run result, namely the "data" key in the API response
+
+    Raises:
+        - prefect.engine.signals.FAIL: when the response code is != 200
+    """
+    data = additional_args if additional_args else {}
+    data["cause"] = cause
+    trigger_request = requests.post(
+        url=__DBT_CLOUD_TRIGGER_JOB_API_ENDPOINT_V2.format(
+            accountId=account_id, jobId=job_id
+        ),
+        headers={"Authorization": f"Bearer {token}"},
+        data=data,
+    )
+
+    if trigger_request.status_code != 200:
+        raise FAIL(message=trigger_request.reason)
+
+    return trigger_request.json()["data"]
+
+
+def get_job_run(
+    account_id: int, token: str, run_id: int, max_attempts: int, wait_interval: int
+) -> dict:
+    """
+    Get a dbt Cloud job run
+
+    Args:
+        - account_id (int): dbt Cloud account ID
+        - token (string): dbt Cloud token
+        - run_id (int): dbt Cloud job run ID
+        - max_attempts: the maximum number of calls to make to the Get Job Run API
+        - wait_interval: the number of seconds to wait between API calls
+
+    Returns:
+        - The job run result, namely the "data" key in the API response
+
+    Raises:
+        - prefect.engine.signals.FAIL: if "finished_at" is not None and the result status != 10
+    """
+    attempts = 1
+    while not max_attempts or attempts <= max_attempts:
```

Comment:
At first, I read this as retry attempts but this is actually the maximum amount of time you'll wait for the job to be finished. I'd clarify that this function will fail immediately if any of the API requests _fails_. Perhaps instead of `max_attempts` you should have `max_wait_time` which is more meaningful to the user instead of considering the combination of both `max_attempts` and `wait_interval`.

Note this would reflect into your task-level settings as well.

Reply:
I think this function also ought to be renamed to reflect that it will wait for job completion.